puppet/modules/freight/manifests/user.pp

```diff
@@ -70,6 +70,7 @@
   include apache::mod::mime
 
   web::vhost { $vhost:
+    serveraliases => [ "${vhost}-backend.theforeman.org", "${vhost}.${facts['networking']['fqdn']}" ],
```

Comment:
I thought a bit more about this, and here is a tiny braindump:
* Today we configure the CDN to talk to `web01` and set the `Host` header to `deb.theforeman.org`, etc to request the "right" content.
* I *want* the CDN to talk to `deb.<the backend machine>` (and `stagingdeb.…`, etc) as the end goal, as this allows me to
  * select the backend at the CDN level (no DNS involved, as today)
  * drop `deb.tfm.o` etc DNS names from the cert `<the backend machine>` has (as those do not point at that machine directly)
* This means that as the end result, all I need are `${archive}.${backend}` vhosts (so `deb.repo-deb01.osuosl.theforeman.org`, etc)
* To ease the transition, it's beneficial to keep `deb.tfm.o` and similar aliases until the CDN is reconfigured and verified.

TL;DR:

Step 1:
```puppet
 web::vhost { $vhost:
  servername => "${vhost}.${facts['networking']['fqdn']}",
  serveraliases => [ "${vhost}.theforeman.org" ],
  …
}
```

reconfigure CDN

Step 2:
```puppet
 web::vhost { $vhost:
  servername => "${vhost}.${facts['networking']['fqdn']}",
  …
}
```

(need to set `servername` explicitly as `web::vhost` otherwise uses `"${title}.theforeman.org"`)

Reply:
According to your step 1, you should be setting `servername` ?

Reply:
> According to your step 1, you should be setting `servername` ?

In this PR? Yes.

Reply:
I think I misunderstood -- your proposal looks good to me. I think you can update the PR to reflect that.